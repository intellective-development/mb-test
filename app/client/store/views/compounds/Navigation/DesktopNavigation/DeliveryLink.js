// @flow
import * as React from 'react';
import bindClassNames from 'shared/utils/bind_classnames';
import { connect } from 'react-redux';
import { address_selectors } from 'store/business/address';
import { ui_actions } from 'store/business/ui';

import { MBText, MBTouchable } from '../../../elements';
import DeliveryInfo from '../shared/DeliveryInfo';

import styles from './DeliveryLink.scss';

const cn = bindClassNames(styles);

type DeliveryLinkProps = {
  showDeliveryInfoModal: () => void,
  has_delivery_address: boolean
};

const DeliveryLink = ({has_delivery_address, showDeliveryInfoModal}: DeliveryLinkProps) => {
  const handleClick = () => {
    showDeliveryInfoModal(location.pathname);
  };

  return (
    <MBTouchable
      role="button"
      onClick={handleClick}
      className={cn('cmDeliveryLink', {cmDeliveryLink__NoAddress: !has_delivery_address})}>
      <DeliveryInfo />
      <MBText.Span className={cn('cmDeliveryLink_Change', {cmDeliveryLink_Change__NoAddress: !has_delivery_address})}>
        Change
      </MBText.Span>
    </MBTouchable>
  );
};


const DeliveryLinkSTP = (state) => ({has_delivery_address: address_selectors.hasDeliveryAddress(state)});
const DeliveryLinkDTP = {showDeliveryInfoModal: ui_actions.showDeliveryInfoModal};
const DeliveryLinkContainer = connect(DeliveryLinkSTP, DeliveryLinkDTP)(DeliveryLink);

export default DeliveryLinkContainer;
