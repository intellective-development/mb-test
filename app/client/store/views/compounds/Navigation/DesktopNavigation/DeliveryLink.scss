@import "~client/store/views/style/variables";
@import "./_variables";

.cmDeliveryLink {
  display: flex;
  align-items: center;
  padding: 0 $standard-gutter;

  transition: $nav-hover-transition;

  &:hover {
    @include nav-hover-state;

    .cmDeliveryLink_Change {
      visibility: visible;
      opacity: 1;
    }
  }

  @media only screen and (max-width: $nav-contract-max-width) {
    padding-left: 0;
  }
}

.cmDeliveryLink__NoAddress{
  // to balance the padding out, we add some extra whitespace (12px corresponds to the margin on the delivery icon)
  padding-right: $standard-gutter + 12px;
}

.cmDeliveryLink_Change {
  @include font-size(13);
  padding-left: 32px;
  color: $brand-red;
  font-weight: $heavy-font-weight;
  visibility: hidden; // it will only show up when its parent is being hovered

  // sync this button appearing/disapearing with the .delivery_link background
  transition: $nav-hover-transition;
  transition-property: all;
  opacity: 0;
}

.cmDeliveryLink_Change__NoAddress {
  display: none;
}

