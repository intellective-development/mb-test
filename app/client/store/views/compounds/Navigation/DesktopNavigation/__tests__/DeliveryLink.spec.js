
import React from 'react';
import address_factory from 'store/business/address/__tests__/address.factory';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import DeliveryLink from '../DeliveryLink';

describe('DeliveryLink', () => {
  it('renders', () => {
    const initial_state = {
      ...address_factory.stateify(
        address_factory.build({local_id: '10'})
      )
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryLink />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders without address', () => {
    expect(render(
      <TestProvider initial_state={{}}>
        <DeliveryLink />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
