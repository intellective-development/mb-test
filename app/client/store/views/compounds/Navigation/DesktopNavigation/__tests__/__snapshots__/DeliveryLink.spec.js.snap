// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryLink renders 1`] = `
<div
  class="el-touchable cmDeliveryLink"
  role="button"
  tabindex="-1"
>
  <div>
    <img
      alt="on_demand"
      class="cmDeliveryMethodIcon_Multi"
      height="24"
      src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
      width="24"
    />
    <div>
      <span
        class="elMBText elMBText_ResetSpacing"
      >
        Delivery to
      </span>
      <br />
      <span
        class="elMBText elMBText_ResetSpacing"
      >
         123 Jump Street, Apt. 10
      </span>
    </div>
  </div>
  <span
    class="elMBText elMBText_ResetSpacing cmDeliveryLink_Change"
  >
    Change
  </span>
</div>
`;

exports[`DeliveryLink renders without address 1`] = `
<div
  class="el-touchable cmDeliveryLink cmDeliveryLink__NoAddress"
  role="button"
  tabindex="-1"
>
  <div>
    <img
      alt="on_demand"
      class="cmDeliveryMethodIcon_Multi"
      height="24"
      src="/assets/components/compounds/delivery_method_icon/on_demand.svg"
      width="24"
    />
    <div>
      <span
        class="elMBText elMBText_ResetSpacing"
      >
        Delivery to
      </span>
      <br />
      <span
        class="elMBText elMBText_ResetSpacing"
      >
        Add delivery address
      </span>
    </div>
  </div>
  <span
    class="elMBText elMBText_ResetSpacing cmDeliveryLink_Change cmDeliveryLink_Change__NoAddress"
  >
    Change
  </span>
</div>
`;
